### DISCUSSION TITLE
Can't create second site

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3960] [Q&A]
(Note: this discussion follows on from https://github.com/zotonic/zotonic/discussions/3953)

I tried creating a second site, in the same way I successfully created the first site. It doesn't work. 
<img width="766" alt="Screenshot 2024-11-27 at 10 27 39" src="https://github.com/user-attachments/assets/581dfe67-c356-4395-9ccb-b74a69092b72">

The error from the logs

```
2024-11-27 08:24:55 ERROR <0.16172.0> [:] ▸ text="Error in process, exit" result=error reason=undef pid=<0.16172.0> node=zotonic@localhost exit_value=badarg
  Stack:
     os:cmd([103,105,116,32,105,110,'...']) @ "os.erl":443
     zotonic_status_addsite:addsite_check_git/3 @ "/opt/zotonic/apps/zotonic_mod_zotonic_site_management/src/zotonic_status_addsite.erl":123
     mod_zotonic_site_management:event/2 @ "/opt/zotonic/apps/zotonic_mod_zotonic_site_management/src/mod_zotonic_site_management.erl":58
     z_transport:incoming_postback_event/6 @ "/opt/zotonic/apps/zotonic_mod_wires/src/support/z_transport.erl":178
     z_transport:transport/3 @ "/opt/zotonic/apps/zotonic_mod_wires/src/support/z_transport.erl":94
     mod_wires:mqtt:zotonic-transport/+/2 @ "/opt/zotonic/apps/zotonic_mod_wires/src/mod_wires.erl":71
     lists:foreach_1/2 @ "lists.erl":1442
```

The result is an empty folder: `apps_user/test2`. If I delete the folder and try again, I get a different error, which claims that the folder exists (though it doesn't).

**Comment by mworrell** [comment]
That is very weird.  The error basically says that the function `os:cmd/1` is undefined.

Could you try:

```
os:cmd("git -v").
```

On the Erlang command line of your instance?

**Reply by geoidesic** [reply]
```
os:cmd("git --version").
"git version 2.45.2\n"
(zotonic@localhost)2> 
```
Why is it trying to access git though? I haven't added any git credentials for site2.

**Comment by mworrell** [comment]
It is using `git` to do a `git init -q` command, so that the site is prepared to use git later.

Somehow the `os:cmd/1` call failed, due to the core Erlang `os` module not being loaded. Which is kind of weird.

**Reply by geoidesic** [reply]
How can I assist you to debug this so that you can prevent this gotcha for new developers implementing Zotonic?

**Reply by mworrell** [reply]
I am also wondering - as it is a bit strange that the `os:cmd/1` suddenly gives an error...
